### PR TITLE
feat: Persist global selections in local storage

### PR DESCRIPTION
**Pull request by lynnagara**
Persist latest project, environment and time selections in local storage.
These will be applied unless a querystring is provided.

### CODE REVIEW

File: src/sentry/static/sentry/app/stores/globalSelectionStore.jsx [hunk 1]
```diff
@@ -34,6 +40,40 @@ const GlobalSelectionStore = Reflux.createStore({
...
+  loadInitialData(queryParams) {
+    const query = pick(queryParams, Object.values(URL_PARAM));
+    const hasQuery = Object.keys(query).length > 0;
+
+    let globalSelection = getDefaultSelection();
+
+    if (hasQuery) {
+      const parsed = getStateFromQuery(queryParams);
+      globalSelection = {
+        projects: parsed.project || [],
```
Comment (markstory) [inline src/sentry/static/sentry/app/stores/globalSelectionStore.jsx]: Do we know if the consumers of global selection will behave well if a user:

1. Has a project stored in local storage
2. They then lose access to one or more stored projects.

File: src/sentry/static/sentry/app/stores/globalSelectionStore.jsx [hunk 2]
```diff
@@ -34,6 +40,40 @@ const GlobalSelectionStore = Reflux.createStore({
...
+        datetime: {
+          [DATE_TIME.START]: parsed.start || null,
+          [DATE_TIME.END]: parsed.end || null,
+          [DATE_TIME.PERIOD]: parsed.period || null,
+          [DATE_TIME.UTC]: parsed.utc || undefined,
+        },
+      };
+    } else {
+      try {
+        globalSelection = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
```
Comment (markstory) [inline src/sentry/static/sentry/app/stores/globalSelectionStore.jsx]: Should selections be stored per org?

Comment (lynnagara) [reply src/sentry/static/sentry/app/stores/globalSelectionStore.jsx]: Oh, that's a good point, will update this.

File: src/sentry/static/sentry/app/stores/globalSelectionStore.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/stores/globalSelectionStore.jsx]: I'm not sure, but I'll update this to do some validation before putting this into the store

**Comment by lynnagara** [comment]
I don't know why this causes the Discover snapshots to fail

**Review by markstory — APPROVED** [review]
Looks good to me. Outdated projects will cause a validation error and not update the store with incorrect data :+1:

**Pull request merged**